module/VuFindTheme/src/VuFindTheme/Initializer.php

```diff
@@ -440,4 +443,41 @@ protected function updateTranslator($themes)
             }
         }
     }
+
+    /**
+     * Load configured javascript files from mixin.config.php if existing
+     *
+     * @return void
+     */
+    public function loadConfiguredJavascriptFilesFromMixin(): void
+    {
+        $config = $this->serviceManager->get('config');
+        $templatePathStack = $config['view_manager']['template_path_stack'] ?? false;
+        if ($templatePathStack) {
+            /* @var HeadScript $headScript */
+            $headScript = $this->serviceManager->get('ViewHelperManager')
+                    ->get(HeadScript::class);
+
+            foreach ($templatePathStack as $templatePath) {
+                $merged = $this->tools->getMergedConfig(
+                    $templatePath . '/../mixin.config.php'
```

Comment:
getMergedConfig is not intended to accept a filename as a parameter, but rather a configuration key to retrieve -- so, for example, you could call it with 'js' if you want to get the final merged list of Javascript settings from all the themes and mixins, since that's the array key where Javascript files are listed.

I also notice that this code seems to be assuming that JS files will always be listed as filenames, but that overlooks the array format of configuration that's been supported since release 8.0.

On further reflection, is it possible that all of this logic should be moved out of the Initializer and into the [SetupThemeResources helper](https://github.com/vufind-org/vufind/blob/dev/module/VuFindTheme/src/VuFindTheme/View/Helper/SetupThemeResources.php)? That's where we actually load the scripts into the appropriate header or footer location, so would that be the best time to pre-process non-web-accessible scripts?

Reply:
Thanks for your remarks. GetMergedConfig should indeed keep it's expected behavior. Perhaps by setting the vendor theme as [mixin](https://github.com/vufind-org/vufind/blob/dev/module/VuFindTheme/src/VuFindTheme/ThemeInfo.php#L256) of the current theme set.

But do I understand you correctly and we should not use GetMergedConfig AND the Initializer anymore? I thought it would be neccessary to use the asset pipeline. Or did we loose that option after migrating from InjectTemplateListenerFactory already? SetupThemeResources and the ResourceContainer are quite low level and to move "loadConfiguredJavascriptFilesFromMixin" there seems unhandily, although parsing package resources could be happen only once.

Reply:
I believe it may not be necessary to use getMergedConfig, and to reduce the number of changes to the Initializer. My thought is this: the purpose of the ResourceContainer is to collect information about resources necessary to provide them to the user. The purpose of SetupThemeResources is to configure the resource-related view helpers based on the content of the ResourceContainer. All of the asset pipeline logic is handled internally by the view helpers.

Since ResourceContainer already represents information about resources as arrays, it should be possible to adjust the Initializer code that populates the ResourceContainer to provide enough context (e.g. theme base path as a second parameter to addJs) to include a "full path to file" element in the information array. With this information present, then when SetupThemeResources loads data into the view helpers, it can make appropriate compensation for files that are identified as existing outside of the web-accessible area based on these paths. I don't think you need to create mixin-specific logic to figure this out.

Of course, it's possible that I'm overlooking something and this will not work -- but it seems like it might be worth considering/investigating. Does that make any sense?

Reply:
Yes, of course, it does. I wasn't quite sure about the details. I will try that. Thank you.